### PR TITLE
10.0 pos orders history imp

### PR DESCRIPTION
**Pull request by GabbasovDinar**

### CODE REVIEW

**Review by ilmir-k — APPROVED** [review]

File: pos_orders_history/static/src/js/screens.js [hunk 1]
```diff
@@ -318,8 +321,36 @@ odoo.define('pos_orders_history.screens', function (require) {
...
             });
             var screen_name = this.gui.get_current_screen();
-            if (order && screen_name === "orders_history_screen") {
-                this.search_order_on_history(order);
+            if (screen_name === "orders_history_screen") {
+                if (order) {
+                    this.search_order_on_history(order);
+                } else {
+                    // send request to server
+                    new Model('pos.order').call('search_read', [[['pos_reference', '=', 'Order ' + code.code]]]).then(function(o) {
```
Comment (Ramil-Mukhametzyanov) [inline pos_orders_history/static/src/js/screens.js]: This part can cause incompatible problems in the non-English interface.
Do not use the word 'Order'.

**Review by Ramil-Mukhametzyanov — CHANGES_REQUESTED** [review]
if using 'Load Specific Orders only'
1. Can't find any orders in the history by searching. I used this example 'Order 00001-001-0001'
2. Scanning '00001-001-0001' will cause the error
    Error: Could not search the Order
    There is no order with this barcode
If this option is not used, the order is successfully found.

**Review by Ramil-Mukhametzyanov — APPROVED** [review]
Understand this feature

File: pos_orders_history/static/src/js/screens.js [hunk 2]
```diff
@@ -318,8 +321,36 @@ odoo.define('pos_orders_history.screens', function (require) {
...
+                            self.gui.show_popup('error',{
+                                'title': _t('Error: Could not search the Order'),
+                                'body': _t('There is no order with this barcode.')
+                            });
+                        }
+                    }, function(err, event) {
+                        event.preventDefault();
+                        console.error(err);
+                        self.gui.show_popup('error',{
+                            'title': _t('Error: Could not search the Order'),
```
Comment (KolushovAlexandr) [inline pos_orders_history/static/src/js/screens.js]:
```suggestion
                            'title': _t('Error: Could not find the Order'),
```

File: pos_orders_history/static/src/js/screens.js [hunk 3]
```diff
@@ -318,8 +321,36 @@ odoo.define('pos_orders_history.screens', function (require) {
...
+                            if (o instanceof Array) {
+                                o = o[0];
+                            }
+                            self.pos.get_order_history_lines_by_order_id(o.id).done(function (lines) {
+                                self.pos.update_orders_history_lines(lines);
+                                self.search_order_on_history(o);
+                            });
+                        } else {
+                            self.gui.show_popup('error',{
+                                'title': _t('Error: Could not search the Order'),
```
Comment (KolushovAlexandr) [inline pos_orders_history/static/src/js/screens.js]:
```suggestion
                                'title': _t('Error: Could not find the Order'),
```

**Review by KolushovAlexandr — APPROVED** [review]

**Review by Ramil-Mukhametzyanov — CHANGES_REQUESTED** [review]
Remove the word 'Order' from the order in the search string that appears in the history after scanning

**Comment by ilmir-k** [comment]
What is the progress on this PR?

**Comment by GabbasovDinar** [comment]
@ilmir-k @Ramil-Mukhametzyanov You can check a new feature https://github.com/it-projects-llc/pos-addons/pull/887/commits/bcd5683816eb80f3b49687baa14ddac5d1302218 and then merge the PR

**Review by Ramil-Mukhametzyanov — APPROVED** [review]

**Pull request merged**